### PR TITLE
adoptopenjdk-icedtea-web: rename from icedtea_web

### PR DESCRIPTION
**Pull request by stefano-m**
###### Motivation for this change

This is a follow up of https://github.com/NixOS/nixpkgs/pull/66422

- rename icedtea_web to adoptopenjdk-icedtea-web to reflect the new governance
- add icedtea_web and icedtea8_web to aliases.nix for backwards compatibility

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @worldofpeace (who reviewed #66422)

### CODE REVIEW

File: pkgs/development/compilers/adoptopenjdk-icedtea-web/default.nix [hunk 1]
```diff
@@ -1,7 +1,7 @@
 { stdenv, fetchFromGitHub, cargo, rustc, autoreconfHook, jdk, glib, xulrunner, zip, pkgconfig, npapi_sdk, bash, bc }:
 
 stdenv.mkDerivation rec {
-  name = "icedtea-web-${version}";
+  name = "adoptopenjdk-icedtea-web-${version}";
```
Comment (worldofpeace) [inline pkgs/development/compilers/adoptopenjdk-icedtea-web/default.nix]:
```suggestion
  pname = "adoptopenjdk-icedtea-web";
```

**Comment by worldofpeace** [comment]
We need to rename the references within nixpkgs as well.

This was the only one
```
pkgs/applications/networking/browsers/firefox/wrapper.nix
6:, jrePlugin, icedtea_web
58:          ++ lib.optional icedtea icedtea_web
```

**Comment by stefano-m** [comment]
> We need to rename the references within nixpkgs as well.

Thanks, I'll fix that.

**Comment by stefano-m** [comment]
force-pushed to fix the little mess I made

**Comment by stefano-m** [comment]
FTR, I wonder whether it still makes sense to keep icedtea-web in firefox since support for Java plugins has been [removed in version 52.0 (released on 7th of March 2017)](https://www.mozilla.org/en-US/firefox/52.0/releasenotes/)?

**Comment by stefano-m** [comment]
One could technically try to build `firefox-esr-52` with icedtea support, but the package is marked as insecure, so before doing

```
nix-build -I nixpkgs=./ -E 'with import <nixpkgs> { }; wrapFirefox firefox-esr-52-unwrapped { cfg = {icedtea = true;}; }'
```

they would also need to add

```nix
  permittedInsecurePackages = [
    "firefox-52.9.0esr"
    "firefox-esr-unwrapped-52.9.0esr"
  ];
```

To their nix/nixos configuration.

File: pkgs/development/compilers/adoptopenjdk-icedtea-web/default.nix [hunk 2]
```diff
@@ -1,14 +1,14 @@
 { stdenv, fetchFromGitHub, cargo, rustc, autoreconfHook, jdk, glib, xulrunner, zip, pkgconfig, npapi_sdk, bash, bc }:
 
 stdenv.mkDerivation rec {
-  name = "icedtea-web-${version}";
+  name = "adoptopenjdk-icedtea-web";
```
Comment (worldofpeace) [inline pkgs/development/compilers/adoptopenjdk-icedtea-web/default.nix]: This is missing the `p`.
```suggestion
  pname = "adoptopenjdk-icedtea-web";
```

Comment (stefano-m) [reply pkgs/development/compilers/adoptopenjdk-icedtea-web/default.nix]: I'm not familiar with this pattern. I've always seen and used `name`, do we need `pname` and `version` then?

Comment (worldofpeace) [reply pkgs/development/compilers/adoptopenjdk-icedtea-web/default.nix]: see https://github.com/NixOS/nixpkgs/blob/master/doc/stdenv.xml#L34

Comment (stefano-m) [reply pkgs/development/compilers/adoptopenjdk-icedtea-web/default.nix]: Thanks for the pointer!

**Comment by worldofpeace** [comment]
Thanks for following up @stefano-m :sparkles:

**Pull request merged**